streaming/base/dataset.py

```diff
@@ -305,6 +368,54 @@ def _download_file(self, basename: str, wait: bool = False) -> str:
         download_or_wait(remote, local, wait, self.retry, self.timeout)
         return local
 
+    def _download_shard_part(self,
+                             shard: int,
+                             partition: Partition,
+                             raw_info: FileInfo,
+                             zip_info: Optional[FileInfo] = None,
+                             compression: Optional[str] = None) -> None:
+        """Download shard data given metadata for the raw and compressed versions of it.
+
+        MDS format uses joint shards (ie, one file per shard). Other formats supported by streaming
+        use split shards (ie, shard data lives in two files per shard: the raw data itself and
+        metadata in a separate file).
+
+        Args:
+            shard (int): Shard ID.
+            partition (Partition): Our rank and worker's partition of the dataset.
+            raw_info (FileInfo): Raw file info.
+            zip_info (FileInfo, optional): Zip file info. Default: ``None``.
+            compression (str, optional): Compression algorithm used for zip_info. Default: ``None``.
+        """
+        # If the local raw file already exists, this is a no-op.
+        raw_filename = os.path.join(self.local, self.split, raw_info.basename)
+        if os.path.isfile(raw_filename):
+            return
+
+        # Determine if we are the doer (process performing the download), or the waiter (one waiting
+        # for it to complete). Multiple processes share a shard when it is split between workers.
+        wait = shard not in partition.shards_to_download
+
+        # Is compression used?
+        if zip_info:
```

Comment:
I am assuming `zip` keyword name means a compressed information. For example, `zip_filename` means `compressed file name` which can be `gz`, `zstd`, etc right? Then can we not used `zip` keyword for `compression` since it is misleading that we are looking for `zip` extension. Maybe use ``compressed_info` and `compressed_filename` or maybe other names?

Reply:
- i originally tried that, but there are a lot of variable names that need to be prefixed accordingly to whichever naming scheme we use and it got completely unwieldly due to length/verbosity
- if we use `compressed_`, is the opposite file `uncompressed_` ('before compression' but also 'never to have been compressed') or `decompressed_` ('after undoing compression')
- the z in gz, zstd, etc. all stand for zip
- also tried variable name scheme `cmp_` and `dec_`
- after much gnashing, went with the convention `zip_` and `raw_` which seemed clearest

Reply:
Sure. sounds good to me.